### PR TITLE
bookinfo: propagate all tracing headers

### PR DESCRIPTION
**Pull request by brianwolfe**
Update bookinfo to propagate all headers that Istio can use for context propagation.

This will let us:
1. Write more thorough e2e tests for tracing.
2. Demonstrate different context propagation in documentation and examples.

[ ] Configuration Infrastructure
[X] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: samples/bookinfo/src/productpage/productpage.py [hunk 1]
```diff
@@ -182,17 +177,66 @@ def getForwardHeaders(request):
...
+        # Stackdriver Istio configurations. Commented out since they are
+        # propagated by the OpenTracing tracer above.
+        # 'x-b3-traceid',
+        # 'x-b3-spanid',
+        # 'x-b3-parentspanid',
+        # 'x-b3-sampled',
+        # 'x-b3-flags',
+
+        # Application-specific headers to forward.
+        'end-user',
```
Comment (frankbu) [inline samples/bookinfo/src/productpage/productpage.py:224]: I don't think this should be an incoming header here - it's extracted from the session header (line 178, above)

Comment (brianwolfe) [reply samples/bookinfo/src/productpage/productpage.py:224]: ah, you are right. Thanks!

Comment (frankbu) [reply samples/bookinfo/src/productpage/productpage.py:224]: Actually, maybe it's OK. If they are both set, the end-user header will override the session user, but that should never happen anyway.

**Review by frankbu — APPROVED** [review]

File: samples/bookinfo/src/productpage/productpage.py (hunk 1 above)
Comment (brianwolfe) [reply samples/bookinfo/src/productpage/productpage.py]: Removed, I think it's clearer not to double-handle it.

**Pull request merged**